cupy/core/include/cupy/carray.cuh

```diff
@@ -179,6 +179,7 @@ public:
   static const int ndim = _ndim;
 private:
   T* data_;
+  bool c_contiguous;
```

Comment:
I wonder if data alignment in `CArray` matters or not? Could you try moving  this flag to the end of private data members and see if timing changes?

Reply:
let me give it a try again but I had problems with it before

Reply:
You're the boss dude.
 down to `1747.367`

Reply:
changed types to enforce alignment, if defined after the shape & strides array probably the value is not written to the same memory location due to some memory layout differences between nvrtc & gcc. I am not 100% sure of this but I can't find another explanation.

Reply:
I have no idea why we have the alignment issue here...Looks like the struct should be 8 byte aligned (due to `ptrdiff_t`)? Sorry I didn't help 😅